crypto/fipsmodule/evp/digestsign.c

```diff
@@ -78,8 +79,19 @@ DEFINE_LOCAL_DATA(struct evp_md_pctx_ops, EVP_MD_pctx_ops) {
 }
 
 static int uses_prehash(EVP_MD_CTX *ctx, enum evp_sign_verify_t op) {
-  return (op == evp_sign) ? (ctx->pctx->pmeth->sign != NULL)
-                          : (ctx->pctx->pmeth->verify != NULL);
+  // Pre-hash modes of ML-DSA that uses an external mu calculation differs from
+  // other signing algorithms, so we specifically check for NIDs of type NID_MLDSAXX.
```

Comment:
> Pre-hash modes of ML-DSA that uses an external mu calculation differs from other signing algorithms

in a way that isn't meaningful to the intent of this function? ML-DSA now has a pre-hash, so it's not clear why it's an exception to this function's semantic

Reply:
The external mu calculation is precluded using `uses_prehash` from calculating the digest in `EVP_Digest<Sign|Verify>` because:
- it's pure ML-DSA with the digest calculation extracted out and left to the caller to perform.
- it's not a regular hash call on the message only as in the cases of ECDSA and RSA; it includes concatenating the public key with other bytes then the message.
- When FIPS 204 HashML-DSA is supported, it will have its own NIDs and will not be precluded from `uses_prehash`, because calculating the digest of the message first is a step in it.